Modules/Core/Common/include/itkBoolean.h

```diff
@@ -0,0 +1,49 @@
+/*=========================================================================
+ *
+ *  Copyright NumFOCUS
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0.txt
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ *=========================================================================*/
+#ifndef itkBoolean_h
+#define itkBoolean_h
+
+namespace itk
+{
+
+/** The type alias `Boolean` provides an alternative to `bool`. `std::vector<bool>` is not thread safe due
+ * to the possibility of multiple bits being packed together in the same memory location.
+ * std::vector<Boolean> does not have such a space optimization. std::vector<Boolean> is
+ * semantically like `std::vector<bool>`, but unlike `std::vector<bool>`, it does "avoid data races when the contents of
+ * the contained object in different elements in the same container [...] are modified concurrently", according to the
+ * C++ Standard, section [container.requirements.dataraces], "Container data races".
+ *
+ */
+
+class Boolean
+{
+public:
+  constexpr Boolean(bool v = false)
```

Comment:
I see your proposed Boolean class is different from `bool`, as a declared local `bool` may be uninitialized, while your proposal would make a default-constructed `Boolean` false. That's an intended difference, right? Just to be sure!

    bool x; // uninitialized;  
    itk::Boolean y; // false

Reply:
It seems safer this way, but I am open to suggestions.

Reply:
> It seems safer this way, but I am open to suggestions.

It's OK to me. Just want to be sure that it's an intended difference between itk::Boolean and `bool`!

Reply:
Of course `{}` does the job for both of them:

    bool x2{}; // false  
    itk::Boolean y2{}; // false

Reply:
Most relevantly, both `std::vector<bool>(n)` and `std::vector<itk::Boolean>(n)` represent `n` elements, initialized by `false`. So that's also OK!